tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py

```diff
@@ -36,118 +39,175 @@ def _batch_normalization(expander, gamma, beta, x, mean, var, eps, test):
-            if isinstance(self.axis, int):
-                aggr_axes = self.axis,
+            if isinstance(aggr_axes, int):
+                aggr_axes = aggr_axes,
             shape = self.input_shape
             param_shape = tuple(
-                s
-                for i, s in enumerate(shape)
-                if i not in aggr_axes
-            )
-            self.expander = tuple(
+                s for i, s in enumerate(shape) if i not in aggr_axes)
+            expander = tuple(
                 None if i in aggr_axes else slice(None)
-                for i in range(len(shape))
-            )
-
-        options = {}
-        if hasattr(self, 'axis'):
-            options['axis'] = self.axis
-        self.link = links.BatchNormalization(
-            param_shape, dtype=self.dtype, **options)
-        gamma = self.link.gamma.data
-        gamma[...] = numpy.random.uniform(.5, 1, gamma.shape)
-        beta = self.link.beta.data
-        beta[...] = numpy.random.uniform(-1, 1, beta.shape)
-        self.link.cleargrads()
-
-        self.gamma = gamma.copy()[self.expander]  # fixed on CPU
-        self.beta = beta.copy()[self.expander]   # fixed on CPU
-
-        self.x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-        self.gy = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-
-        if self.test:
-            self.mean = numpy.random.uniform(
-                -1, 1, param_shape).astype(self.dtype)
-            self.var = numpy.random.uniform(
-                0.5, 1, param_shape).astype(self.dtype)
-            self.link.avg_mean[...] = self.mean
-            self.link.avg_var[...] = self.var
+                for i in range(len(shape)))
+        elif hasattr(self, 'ndim'):
+            aggr_axes = (0,) + tuple(six.moves.range(2, self.ndim + 2))
+            shape = (5, 3) + (2,) * self.ndim
+            param_shape = shape[1]
+            expander = (None, Ellipsis) + (None,) * self.ndim
         else:
-            self.mean = self.x.mean(axis=aggr_axes)
-            self.var = self.x.var(axis=aggr_axes)
-        self.check_forward_optionss = {'atol': 1e-4, 'rtol': 1e-3}
-        self.check_backward_optionss = {'atol': 1e-4, 'rtol': 1e-3}
+            assert False
+
+        self.aggr_axes = aggr_axes
+        self.shape = shape
+        self.param_shape = param_shape
+        self.expander = expander
+        self.finetune = False
+        self.mean = None
+        self.var = None
+        self.eps = 2e-5
+
+        self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-4, 'rtol': 1e-3}
         if self.dtype == numpy.float16:
-            self.check_forward_optionss = {'atol': 1e-3, 'rtol': 1e-2}
-            self.check_backward_optionss = {'atol': 5e-1, 'rtol': 1e-1}
+            self.check_forward_options = {'atol': 1e-2, 'rtol': 1e-1}
+            self.check_backward_options = {'atol': 5e-1, 'rtol': 1e-1}
+
+    def before_test(self, test_name):
```

Comment:
I think this can be done in `setUp`.

Reply:
`before_test` is used for resampling.

Reply:
Discussed: 
* Resampling is not needed in this particular test. Currently we don't come up with links that need such resampling. That can be implemented by resampling in `create_link` if such links are found. 
* Call `before_test` once before each test.

Reply:
Fixed as disccused.